src/backend/gporca/libgpos/include/gpos/common/CAutoP.h

```diff
@@ -38,10 +38,9 @@ class CAutoP : public CStackObject
 	// actual element to point to
 	T *m_object;
 
-	// hidden copy ctor
-	CAutoP<T>(const CAutoP &);
-
 public:
+	CAutoP<T>(const CAutoP &) = delete;
```

Comment:
The change fix this warning:
```sh
/home/gpadmin/gpdb_src/src/backend/gporca/libgpos/server/src/unittest/gpos/common/CAutoPTest.cpp: In static member function 'static gpos::GPOS_RESULT gpos::CAutoPTest::EresUnittest_Basics()':
/home/gpadmin/gpdb_src/src/backend/gporca/libgpos/server/src/unittest/gpos/common/CAutoPTest.cpp:67:16: error: implicitly-declared 'gpos::CAutoP<char>& gpos::CAutoP<char>::operator=(const gpos::CAutoP<char>&)' is deprecated [-Werror=deprecated-copy]
   67 |         asz3 = asz2;
      |                ^~~~
In file included from /home/gpadmin/gpdb_src/src/backend/gporca/libgpos/server/src/unittest/gpos/common/CAutoPTest.cpp:15:
/home/gpadmin/gpdb_src/src/backend/gporca/libgpos/include/gpos/common/CAutoP.h:42:9: note: because 'gpos::CAutoP<char>' has user-provided 'gpos::CAutoP<T>::CAutoP(const gpos::CAutoP<T>&) [with T = char]'
   42 |         CAutoP<T>(const CAutoP &);
      |         ^~~~~~~~~
```
I took this change from GPDB 7.

Explanation: 
By marking the copy constructor as `= delete`, we explicitly prevent any copy operations on CAutoP objects. This change clarifies that `CAutoP` does not support copying, thus avoiding the deprecated implicit copy assignment operator generation.
Moving the deleted copy constructor to the public section makes the restriction clear and prevents unintended usage

Reply:
Can we delete the constructor instead of marking it?
```c++
template <class T>
class CAutoP
{
protected:
	T *m_object;
//	CAutoP<T>(const CAutoP &);
public:
	explicit CAutoP<T>() : m_object(nullptr){}
	explicit CAutoP<T>(T *object) : m_object(object){}

	CAutoP<T> const & operator=(T *object)
	{
		m_object = object;
		return *this;
	}

};

int main(int argc, char** argv) {
	CAutoP<char> asz2, asz3;
	asz2 = *argv;
	asz3 = asz2;
	return 0;
}
```

g++ issued the same warning before commenting the ctor.
```
$ g++ 2.cc -std=c++11 -Wextra 
2.cc: In function ‘int main(int, char**)’:
2.cc:22:9: warning: implicitly-declared ‘CAutoP<char>& CAutoP<char>::operator=(const CAutoP<char>&)’ is deprecated [-Wdeprecated-copy]
   22 |  asz3 = asz2;
      |         ^~~~
2.cc:6:2: note: because ‘CAutoP<char>’ has user-provided ‘CAutoP<T>::CAutoP(const CAutoP<T>&) [with T = char]’
    6 |  CAutoP<T>(const CAutoP &);
      |  ^~~~~~~~~
```

Reply:
Looks like you are right. I removed unused ctor from CSyncHashtableTest.h. And all works fine. Thanks!

Reply:
I think the hidden ctor
```cpp
	// hidden copy ctor
	CAutoP<T>(const CAutoP &);
```
 in old code was introduced to prevent objects of this class from being copied. By declaring the copy constructor in the private section of the class without defining it, the class prevents copying. We can remove it, but in this case we will miss this protection.
Using = delete is a more explicit and [modern](https://learn.microsoft.com/en-us/cpp/cpp/explicitly-defaulted-and-deleted-functions?view=msvc-170) approach to save this protection.